openedx_external_enrollments/external_enrollments/icc_external_enrollment.py

```diff
@@ -0,0 +1,191 @@
+            }
+            response = requests.post(
+                url=settings.ICC_BASE_URL,
+                data=request_data,
+            )
+            log_details["request_payload"] = request_data
+        except Exception as error:  # pylint: disable=broad-except
+            LOG.error("Failed to retrieve ICC user. Reason: %s", str(error))
+            log_details["request_payload"]["wstoken"] = "icc-api-token"
+            log_details["response"] = {"error": "Failed to retrieve ICC user. Reason: " + str(error)}
+            EnrollmentRequestLog.objects.create(  # pylint: disable=no-member
+                request_type=str(self),
+                details=log_details,
+            )
+        else:
+            icc_user = self._get_icc_user_from_xml_response(response, "get_user")
+            icc_user = _validate_icc_user(icc_user)
+
+        return icc_user
+
+    def _create_icc_user(self, data, duplicated_username):
+        """
+        Method that creates a user in the ICC database based in the  current user logged in data.
+        Also takes care of duplicated usernames.
+        """
+        log_details = {
+            "url": settings.ICC_CREATE_USER_API_FUNCTION,
+        }
+        icc_user = {}
+
+        try:
+            user, _ = get_user(email=data.get("user_email"))
+            request_data = {
+                "wstoken": settings.ICC_API_TOKEN,
+                "wsfunction": settings.ICC_CREATE_USER_API_FUNCTION,
+                "users[0][username]": user.username + self._get_random_string(
+                    self.USERNAME_SUFFIX_LENGTH) if duplicated_username else user.username,
+                "users[0][password]": self._get_random_string(settings.ICC_HASH_LENGTH),
+                "users[0][firstname]": user.first_name,
+                "users[0][lastname]": user.last_name,
+                "users[0][email]": user.email,
+                "users[0][auth]": "saml2",
+            }
+
+            response = requests.post(
+                url=settings.ICC_BASE_URL,
+                data=request_data,
+            )
+            log_details["request_payload"] = request_data
+        except Exception as error:  # pylint: disable=broad-except
+            LOG.error("Failed to create ICC user. Reason: %s", str(error))
+            log_details["request_payload"]["wstoken"] = "icc-api-token"
+            log_details["response"] = {"error": "Failed to create ICC user. Reason: " + str(error)}
+            EnrollmentRequestLog.objects.create(  # pylint: disable=no-member
+                request_type=str(self),
+                details=log_details,
+            )
+        else:
+            icc_user = self._get_icc_user_from_xml_response(response, "create_user")
+
+        return icc_user
+
+    def _validate_icc_user(self, icc_user):
+        """Ensure the user creation avoiding failure for username duplicates"""
+        duplicated_username = False
+        call_counter = 0
```

Comment:
Please add a blank line after this line. @ivanvgh

Reply:
Done.